cmake/ProjectTcMalloc.cmake

```
@@ -0,0 +1,37 @@
+include(ExternalProject)
+
+set(TCMALLOC_CONFIG ./autogen.sh COMMAND ./configure  --disable-shared CXXFLAGS=-DHAVE_POSIX_MEMALIGN_SYMBOL=1 --enable-frame-pointers --disable-cpu-profiler --disable-heap-profiler --disable-heap-checker --disable-debugalloc --enable-minimal)
+
+if (APPLE)
+    set(TCMALLOC_MAKE make)
+else()
+    set(TCMALLOC_MAKE make)
+endif()
+
+ExternalProject_Add(gptcmalloc
+    PREFIX ${CMAKE_SOURCE_DIR}/deps/
+    DOWNLOAD_NAME master.zip
+    DOWNLOAD_NO_PROGRESS 1
+    URL https://github.com/gperftools/gperftools/archive/master.zip
+    URL_HASH SHA256=2ad5c0685585e1ea818c4a254d97efe6d63644ec32376035d1702e5123c7c94b
+    BUILD_IN_SOURCE 1
+    LOG_CONFIGURE 1
+    LOG_BUILD 1
+    LOG_INSTALL 1
+    CONFIGURE_COMMAND ${TCMALLOC_CONFIG} 
+    BUILD_COMMAND ${TCMALLOC_MAKE}
+    INSTALL_COMMAND ""
+    BUILD_BYPRODUCTS <SOURCE_DIR>/.libs/libtcmalloc_minimal.a
+)
+
+ExternalProject_Get_Property(gptcmalloc SOURCE_DIR)
+add_library(tcmalloc STATIC IMPORTED GLOBAL)
```

```suggestion
add_library(TCMalloc STATIC IMPORTED GLOBAL)
```
what about using it's official name, as same as others